### PR TITLE
Add MockRepo Class and use that in all tests

### PR DESCRIPTION
**Pull request by GQDeltex**
Add MockRepo Class and use that in all tests

### CODE REVIEW

File: backend/src/tools/memdb.mock.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    };
+  }
+
+  async clearRepo(): Promise<void> {
+    await this.repo.clear();
+    await this.source.destroy();
+  }
+
+  async destroyRepo(): Promise<void> {
+    dropDatabase(this.options);
```
Comment (Yuerino) [inline backend/src/tools/memdb.mock.ts:64]: missing await for async function

File: backend/src/tools/memdb.mock.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          initialDatabase: configService.get('DB_NAME'),
+          ifNotExist: true,
+          synchronize: true,
+        };
+        await createDatabase(this.options);
+        this.source = await new DataSource(dataSourceOptions).initialize();
+        this.repo = await this.source.getRepository(this.entity);
+        // Just for safety, don't know if really needed
+        for (let i = 0; i < 10; i++) {
+          if (this.repo.manager.connection.isConnected) break;
```
Comment (Yuerino) [inline backend/src/tools/memdb.mock.ts:48]: `isConnected` is deprecated, use `isInitialized` instead

**Review by Yuerino — CHANGES_REQUESTED** [review]
Other that that, everything looks good, test passed. Amazing work for the test, thank you.

File: backend/src/tools/memdb.mock.ts (hunk 2 above)
Comment (GQDeltex) [reply backend/src/tools/memdb.mock.ts]: [e27e9b5](https://github.com/GQDeltex/ft_transcendence/pull/128/commits/e27e9b542eca3aec7b1e7d3ea2abdd5e22cd1383)

File: backend/src/tools/memdb.mock.ts (hunk 1 above)
Comment (GQDeltex) [reply backend/src/tools/memdb.mock.ts]: [e27e9b5](https://github.com/GQDeltex/ft_transcendence/pull/128/commits/e27e9b542eca3aec7b1e7d3ea2abdd5e22cd1383)

**Review by Yuerino — APPROVED** [review]
LGTM

**Pull request merged**